src/Aspire.Hosting/ApplicationModel/DockerfileBuildAnnotation.cs

```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+internal class DockerfileBuildAnnotation(string contextPath, string dockerfilePath, string? stage) : IResourceAnnotation
+{
+    public string ContextPath => contextPath;
+    public string DockerfilePath = dockerfilePath;
+    public string? Stage => stage;
+    public Dictionary<string, object> BuildArguments { get; } = new();
```

Comment:
Why not use the existing `DockerBuildArg` object?

Reply:
`DockerBuildArg` doesn't give us much (anything) beyond `KeyValuePair<TKey, TValue>`. If this annotation type was public I might be tempted but in this case it would just result in spinning up a bunch of unnecessary objects.

`DockerBuildArg` is also interesting in the context of `PublishAsDockerfile` because you could pass in duplicate args with that API.

`PublishAsDockerfile` is due some TLC after this change goes in.